proto/terra/treasury/v1beta1/treasury.proto

```diff
@@ -26,13 +26,15 @@ message Params {
   uint64 window_short     = 5 [(gogoproto.moretags) = "yaml:\"window_short\""];
   uint64 window_long      = 6 [(gogoproto.moretags) = "yaml:\"window_long\""];
   uint64 window_probation = 7 [(gogoproto.moretags) = "yaml:\"window_probation\""];
-  
+
   string burn_tax_split = 8 [
     (gogoproto.moretags)   = "yaml:\"burn_tax_split\"",
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
     (gogoproto.nullable)   = false
   ];
 
+  repeated string burn_tax_whitelist = 9
```

Comment:
Instead of putting it in params, how about using it as a genesis state? I think it will be easy to understand if you look at the [validators of the staking module](https://github.com/cosmos/cosmos-sdk/blob/74d7a0dfcd9f47d8a507205f82c264a269ef0612/proto/cosmos/staking/v1beta1/genesis.proto#L29). It can be managed in key-value.

advantages
- We can process addresses one by one when adding or deleting a whitelist as a proposal.
- We don't need to go through a loop each time to check whether it is whitelisted or not.

additional work
- Need to work on the proposal handler.
- Needs some abci and keeper work on genesis and whiltelist management.

Reply:
I don't really understood what the principal difference is. Except that the genesis parameters are... Well... Part of the genesis. Aren't the keeper parameters are managed in a key:value fashion?

Reply:
> I don't really understood what the principal difference is. Except that the genesis parameters are... Well... Part of the genesis.

If it's not `params` and just stored in the keeper, there's a problem with fork using export. If you include it as the genesis state, export is possible, and it can be maintained even if fork through export. I thought this was similar to managing the validator list.
When stored in the keeper, it affects the apphash. In other words, the whitelist affects the state of the chain. 

> Aren't the keeper parameters are managed in a key:value fashion?

`params` is stored as key:value, but whitelist is stored as array. If the whitelist address can be stored as a key:value, finding whether it belongs to the whitelist can be performed in O(1) instead of O(N).

> I also thought the keeper parameters could be handled by governance?

I thought if we added a proposal-handler, we could use the gov to manage the keeper's parameters rather than the params.

Reply:
It doesnt really make a lot of sense to worry about being able to export if we cannot import. So while I understand the idea  around adding it to the genesis we would need to resolve the issues in the Cosmos/Tendermint DB architecture to leverage the value of doing so?

It only affects the apphash if it actually goes into the DB store.  Keeping a hardcoded array in the keeper will not have any impact on the app hash as I understand it, unless something automagically ensures to include the params as part of the app hash generation for block headers?

Reply:
I think we can add a map instead of an array, as long as we don't iterate over a map, there is nothing to worry about

Reply:
Design for clean substore whitelist address as suggested by @JoowonYun 

1. add a substore for whitelist address. <k,v> will be <whitelist address, byte>. This substore will have following func:
- SetAddress() will set a whitelist address
- RemoveAddress() will remove a whitelist address
- HasAddress() will check if a whitelist address exists or not

2. add a proposal handler for setting, removing whitelist address
<img width="727" alt="Screenshot 2023-01-20 at 00 22 50" src="https://user-images.githubusercontent.com/23058164/213516565-0c73fd49-82da-4bf5-8ef1-0da41c7bf46f.png">

3. at WhitelistHeight, we SetAddress() of all whitelist addresses array.

4. code for migrating store of treasury to include substore for whitelist address.

Reply:
if we were to continue with one param contains an array, every time we have to add or remove address through gov, we have to change the whole array which to me is not every elegent.

Reply:
I kinda figured that as a tradeoff between "hardcoded in ante handler" and "having it as gov parameter" - I would be happy to have a map. But I think lookup times for a map is still O(N)

Reply:
Whenever faced with two choices, select the path that makes future change easier. :) So I think we can invest the work in this as I am sure this is not the last time we will have to implement a "parameter stub" :)

Reply:
I can implement this quickly as I have done this before so won't take much time